feat(extra-natives/five): train movement natives

Ehbw opened the pull request:
### Goal of this PR

Introduce a set of getters + a setter for several aspects of train movements. Allows for scripts to get information about the trains current speed, cruise speed, direction and state (e.g. moving, stopping at station, stopped at station and driving away from station) which was previously not possible 

### How is this PR achieving the goal

By creating GET_TRAIN_DIRECTION, GET_TRAIN_STATE, GET_TRAIN_CRUISE_SPEED, GET_TRAIN_SPEED and SET_TRAIN_STATE (to allow for scripts to change the trains current state)

### This PR applies to the following area(s)

FiveM, Natives

### Successfully tested on

**Game builds:** 1604, 3258

**Platforms:** Windows

### Checklist

- [x] Code compiles and has been tested successfully.
- [x] Code explains itself well and/or is documented.
- [x] My commit message explains what the changes do and what they are for.
- [x] No extra compilation warnings are added by these changes.

### Fixes issues

dalekenium-cfx reviewed (approved):
Looks good to me - no major complaints.

Ehbw commented:
Changed SET_TRAIN_STATE to use a function that's used else where in the game code for setting train's new state (it resets a timer used for train states). Also fixed the pattern for Train Speed as for some reason only on b2189 it would return a completely different offset that isn't related to the trains current speed.

I've tested these changes again on all games build (1604 - 3258)
